### PR TITLE
feat(ratelimit): Split the rate-limiting policy into two modules:

### PR DESCRIPTION
**Pull request by brasseld**
1) Quota policy
2) Rate-Limiting policy (throttling)

Closes gravitee-io/issues#245

### CODE REVIEW

File: gravitee-policy-quota/src/main/resources/plugin.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+id=rate-limit
```
Comment (NicolasGeraud) [inline gravitee-policy-quota/src/main/resources/plugin.properties]: wrong id

File: gravitee-policy-quota/src/main/resources/schemas/urn:jsonschema:io:gravitee:policy:quota:configuration:QuotaPolicyConfiguration.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "type" : "object",
+  "id" : "urn:jsonschema:io:gravitee:policy:quota:configuration:QuotaPolicyConfiguration",
+  "properties" : {
+    "async": {
+      "type": "boolean",
+      "default": false,
+      "title": "Non-strict mode (async)",
+      "description": "By activating this option, rate-limiting is applied in an asynchronous way meaning that the distributed counter value is not strict."
```
Comment (NicolasGeraud) [inline gravitee-policy-quota/src/main/resources/schemas/urn:jsonschema:io:gravitee:policy:quota:configuration:QuotaPolicyConfiguration.json]: s/rate-limiting/quota/

File: gravitee-policy-quota/src/main/java/io/gravitee/policy/quota/QuotaPolicy.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+                    executionContext.getAttribute(ExecutionContext.ATTR_SUBSCRIPTION) + KEY_SEPARATOR +
+                    ((resolvedPath != null) ? resolvedPath.hashCode() : "");
+        }
+
+        return (String) executionContext.getAttribute(ExecutionContext.ATTR_SUBSCRIPTION) + KEY_SEPARATOR +
+                RATE_LIMIT_TYPE + KEY_SEPARATOR +
+                ((resolvedPath != null) ? resolvedPath.hashCode() : "");
+    }
+
+    private PolicyResult createLimitExceeded(QuotaConfiguration rateLimitConfiguration) {
```
Comment (NicolasGeraud) [inline gravitee-policy-quota/src/main/java/io/gravitee/policy/quota/QuotaPolicy.java]: rename argument

**Comment by brasseld** [comment]
FP

**Pull request merged**